base_layer/core/src/chain_storage/blockchain_database.rs

```diff
@@ -217,15 +217,18 @@ where B: BlockchainBackend
     pub fn create_chain_header_if_valid(
```

Comment:
This method seems to only be used in tests and should be deleted or moved to a test mod